utilities.py

```diff
@@ -160,6 +160,165 @@ def make_tot_vs_toa_plots(
+    ):
+    min_x = data_df["calibration_code_{}".format(board_a)].min()
+    max_x = data_df["calibration_code_{}".format(board_a)].max()
+    min_y = data_df["calibration_code_{}".format(board_b)].min()
+    max_y = data_df["calibration_code_{}".format(board_b)].max()
+
+    if range_cal is None:
+        range_x = [min_x, max_x]
+        range_y = [min_y, max_y]
+    else:
+        range_x = [
+            max(min_x, range_cal[0]),
+            min(max_x, range_cal[1])
+        ]
+        range_y = [
+            max(min_y, range_cal[0]),
+            min(max_y, range_cal[1])
+        ]
+
+    nbinsx = ceil((range_x[1] - range_x[0]) * 1)  # 1 bins per unit (but it seems plotly uses this more as a suggestion)
+    nbinsy = ceil((range_y[1] - range_y[0]) * 1)  # 1 bins per unit (but it seems plotly uses this more as a suggestion)
+
+    fig = px.scatter(
+        data_df,
+        x="calibration_code_{}".format(board_a),
+        y="calibration_code_{}".format(board_b),
+        labels = {
+            "calibration_code_{}".format(board_a): "Board {} Calibration Code".format(board_a),
+            "calibration_code_{}".format(board_b): "Board {} Calibration Code".format(board_b),
+        },
+        title = "Calibration code correlation between board {} and board {}<br><sup>Run: {}{}</sup>".format(board_a, board_b, run_name, extra_title),
+        opacity = 0.5,
+        range_x=range_x,
+        range_y=range_y,
+    )
+
+    fig.write_html(
+        base_path/'cal_board{}_vs_board{}_scatter.html'.format(board_a, board_b),
+        full_html = full_html,
+        include_plotlyjs = 'cdn',
+    )
+
+    fig = px.density_heatmap(
+        data_df,
+        x="calibration_code_{}".format(board_a),
+        y="calibration_code_{}".format(board_b),
+        labels = {
+            "calibration_code_{}".format(board_a): "Board {} Calibration code [ns]".format(board_a),
+            "calibration_code_{}".format(board_b): "Board {} Calibration code [ns]".format(board_b),
+        },
+        color_continuous_scale="Blues",  # https://plotly.com/python/builtin-colorscales/
+        title = "Calibration code correlation between board {} and board {}<br><sup>Run: {}{}</sup>".format(board_a, board_b, run_name, extra_title),
+        range_x=range_x,
+        range_y=range_y,
+        nbinsx=nbinsx,
+        nbinsy=nbinsy,
+    )
+
+    fig.write_html(
+        base_path/'cal_board{}_vs_board{}.html'.format(board_a, board_b),
+        full_html = full_html,
+        include_plotlyjs = 'cdn',
+    )
+
+def make_calcode_correlation_plot(
+    data_df: pandas.DataFrame,
+    base_path: Path,
+    run_name: str,
+    board_ids: list[int],
+    full_html:bool=False,
+    extra_title: str = "",
+    ):
+
+    cal_dimensions = []
+    cal_labels = {}
+
+    for board_id in board_ids:
+        cal_dimensions += ["calibration_code_{}".format(board_id)]
+        cal_labels["calibration_code_{}".format(board_id)] = "Board {} Calibration code [ns]".format(board_id)
```

Comment:
Fix here. The calibration code does not have units of ns

```suggestion
        cal_labels["calibration_code_{}".format(board_id)] = "Board {} Calibration code".format(board_id)
```

Reply:
I fixed this